gstreamer-rtsp-server/src/auto/r_t_s_p_media.rs

```diff
@@ -0,0 +1,838 @@
+use glib::Value;
+use glib::object::Downcast;
+use glib::object::IsA;
+use glib::signal::SignalHandlerId;
+use glib::signal::connect;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use gst;
+use gst_rtsp;
+use libc;
+use std::boxed::Box as Box_;
+use std::mem;
+use std::mem::transmute;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct RTSPMedia(Object<ffi::GstRTSPMedia, ffi::GstRTSPMediaClass>);
+
+    match fn {
+        get_type => || ffi::gst_rtsp_media_get_type(),
+    }
+}
+
+impl RTSPMedia {
+    pub fn new<P: IsA<gst::Element>>(element: &P) -> RTSPMedia {
+        assert_initialized_main_thread!();
+        unsafe {
+            from_glib_full(ffi::gst_rtsp_media_new(element.to_glib_full()))
+        }
+    }
+}
+
+unsafe impl Send for RTSPMedia {}
+unsafe impl Sync for RTSPMedia {}
+
+pub trait RTSPMediaExt {
+    fn collect_streams(&self);
+
+    fn create_stream<P: IsA<gst::Element>, Q: IsA<gst::Pad>>(&self, payloader: &P, pad: &Q) -> Option<RTSPStream>;
+
+    fn find_stream(&self, control: &str) -> Option<RTSPStream>;
+
+    fn get_address_pool(&self) -> Option<RTSPAddressPool>;
+
+    fn get_base_time(&self) -> gst::ClockTime;
+
+    fn get_buffer_size(&self) -> u32;
+
+    fn get_clock(&self) -> Option<gst::Clock>;
+
+    fn get_element(&self) -> Option<gst::Element>;
+
+    fn get_latency(&self) -> u32;
+
+    fn get_multicast_iface(&self) -> Option<String>;
+
+    //fn get_permissions(&self) -> /*Ignored*/Option<RTSPPermissions>;
+
+    fn get_profiles(&self) -> gst_rtsp::RTSPProfile;
+
+    fn get_protocols(&self) -> gst_rtsp::RTSPLowerTrans;
+
+    fn get_publish_clock_mode(&self) -> RTSPPublishClockMode;
+
+    fn get_range_string(&self, play: bool, unit: gst_rtsp::RTSPRangeUnit) -> Option<String>;
+
+    fn get_retransmission_time(&self) -> gst::ClockTime;
+
+    fn get_status(&self) -> RTSPMediaStatus;
+
+    fn get_stream(&self, idx: u32) -> Option<RTSPStream>;
+
+    fn get_suspend_mode(&self) -> RTSPSuspendMode;
+
+    //fn get_time_provider<'a, P: Into<Option<&'a str>>>(&self, address: P, port: u16) -> /*Ignored*/Option<gst_net::NetTimeProvider>;
+
+    fn get_transport_mode(&self) -> RTSPTransportMode;
+
+    //fn handle_sdp(&self, sdp: /*Ignored*/&mut gst_sdp::SDPMessage) -> bool;
+
+    fn is_eos_shutdown(&self) -> bool;
+
+    fn is_reusable(&self) -> bool;
+
+    fn is_shared(&self) -> bool;
+
+    fn is_stop_on_disconnect(&self) -> bool;
+
+    fn is_time_provider(&self) -> bool;
+
+    fn n_streams(&self) -> u32;
+
+    //fn prepare<'a, P: Into<Option<&'a /*Ignored*/RTSPThread>>>(&self, thread: P) -> bool;
+
+    //fn seek(&self, range: /*Ignored*/&mut gst_rtsp::RTSPTimeRange) -> bool;
+
+    fn set_address_pool(&self, pool: &RTSPAddressPool);
+
+    fn set_buffer_size(&self, size: u32);
+
+    fn set_clock<P: IsA<gst::Clock>>(&self, clock: &P);
+
+    fn set_eos_shutdown(&self, eos_shutdown: bool);
+
+    fn set_latency(&self, latency: u32);
+
+    fn set_multicast_iface(&self, multicast_iface: &str);
```

Comment:
Check if setters are nullable (and add annotations), or mark the getters as not nullable :) This is for everything

Reply:
Done